internal/providers/terraform/azure/mssql_database.go

```diff
@@ -0,0 +1,270 @@
+				},
+			},
+			PriceFilter: &schema.PriceFilter{
+				PurchaseOption: strPtr("Consumption"),
+			},
+		})
+	}
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func mssqlSkuSplit(address, sku string) (string, string, string, error) {
+	if s := strings.Split(sku, "_"); len(s) == 4 && strings.HasPrefix(sku, "GP_S_Gen") {
+		return s[0] + s[1], s[2], s[3], nil
+	}
+	if s := strings.Split(sku, "_"); len(s) == 3 {
+		if strings.HasPrefix(sku, "HS_Gen") ||
+			strings.HasPrefix(sku, "HS_DC") ||
+			strings.HasPrefix(sku, "GP_Gen") ||
+			strings.HasPrefix(sku, "GP_Fsv2") ||
+			strings.HasPrefix(sku, "GP_DC") ||
+			strings.HasPrefix(sku, "BC_Gen") ||
+			strings.HasPrefix(sku, "BC_M") ||
+			strings.HasPrefix(sku, "BC_DC") ||
+			strings.HasPrefix(sku, "BC_DC") {
+			return s[0], s[1], s[2], nil
+		}
+	}
+	return "", "", "", errors.Errorf("Unrecognised MSSQL SKU format for resource %s: %s", address, sku)
+}
+
+func mssqlProductName(tier, family string) string {
+	tierName := map[string]string{
+		"GPS": "General Purpose - Serverless",
+		"HS":  "Hyperscale",
+		"GP":  "General Purpose",
+		"BC":  "Business Critical",
+	}[tier]
+
+	familyName := map[string]string{
+		"Gen5":    "Compute Gen5",
+		"Gen4":    "Compute Gen4",
+		"DC":      "Compute DC-Series",
+		"Fsv2":    "Compute FSv2 Series",
+		"M":       "Compute M Series",
+		"Storage": "Storage",
+		"License": "SQL License",
+	}[family]
+
+	return fmt.Sprintf("/%s - %s/", tierName, familyName)
+}
+
+func mssqlSkuName(cores string, zoneRedundant bool) string {
+	sku := cores + " vCore"
+	if zoneRedundant {
+		sku += " Zone Redundancy"
+	}
+	return sku
+}
+
+func sqlLicenseCostComponent(region, cores, serviceName, tier string) *schema.CostComponent {
+	licenseRegion := "Global"
+	if strings.Contains(region, "usgov") {
+		licenseRegion = "US Gov"
+	}
+	if strings.Contains(region, "china") {
+		licenseRegion = "China"
+	}
+	if strings.Contains(region, "germany") {
+		licenseRegion = "Germany"
+	}
+
+	coresNum, _ := strconv.ParseInt(cores, 10, 64)
+
+	return &schema.CostComponent{
+		Name:           "SQL License",
+		Unit:           "hours",
+		UnitMultiplier: 1,
+		HourlyQuantity: decimalPtr(decimal.NewFromInt(coresNum)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(licenseRegion),
+			Service:       strPtr(serviceName),
+			ProductFamily: strPtr("Databases"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "productName", ValueRegex: strPtr(mssqlProductName(tier, "License"))},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("Consumption"),
+		},
+	}
+}
+
+func mssqlStorageComponent(storageGB *decimal.Decimal, region, serviceName, tier string, zoneRedundant bool) *schema.CostComponent {
+	tierName := map[string]string{
+		"GPS": "General Purpose",
+		"HS":  "Hyperscale",
+		"GP":  "General Purpose",
+		"BC":  "Business Critical",
+	}[tier]
+
```

Comment:
If we go for my `parseMSSQLSku` idea above then this would become:
```go
storageTier := tier
if storageTier == "General Purpose - Serverless" {
  storageTier = "General Purpose"
}
```